server/application-server/src/main/java/de/tum/in/www1/hephaestus/leaderboard/LeaderboardService.java

```diff
@@ -55,8 +59,8 @@ public List<LeaderboardEntry> createLeaderboard() {
             Set<PullRequestReviewDTO> commentSet = new HashSet<>();
 
             user.getReviews().stream()
```

Comment:
I think `user.getReviews` could already support querying by timeframe.

Reply:
So if I understand your idea correctly, you would like to do the filtering in the SQL-Query already?

That's a great idea for performance improvments, I will take a look at it.

Reply:
I tried out a Query like this:
```sql
SELECT u
FROM User u
JOIN FETCH u.pullRequests 
JOIN FETCH u.issueComments
JOIN FETCH u.reviewComments
JOIN FETCH u.reviews re
WHERE re.createdAt between :after and :before
OR re.updatedAt between :after and :before
```
but it doesn't seem to return any results, most likely I'm missing something regarding the format conversion.

Reply:
Managed to find the correct SQL query to filter the dates.
With 63c434d, we now filter directly in the DB.